app/scripts/services/serviceInstances.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("ServiceInstancesService", function($filter,
+                                               $uibModal,
+                                               DataService,
+                                               NotificationsService) {
+
+    function deprovision(apiObject) {
+      var getErrorDetails = $filter('getErrorDetails');
+      var modalScope = {
+        alerts: {
+          deprovision: {
+            type: 'error',
+            message: 'Service \'' + apiObject.spec.serviceClassName + '\' will be deleted and no longer available.'
+          }
+        },
+        detailsMarkup: 'Delete Service?',
+        okButtonText: 'Delete',
+        okButtonClass: 'btn-danger',
+        cancelButtonText: 'Cancel'
+      };
+
+      // TODO: we probably have to handle bindings in here.
+      // either:
+      // - automatically remove the bindings
+      // - tell the user they must manually unbind before continue
+      $uibModal.open({
```

Comment:
You probably want to return the promise here so callers can know when it's complete and if it failed.